napari/layers/points/points.py

```diff
@@ -1175,6 +1200,20 @@ def mode(self, mode):
         self._set_highlight()
         self.events.mode(mode=mode)
 
+    @property
+    def _indices_view(self):
+        return self.__indices_view
+
+    @_indices_view.setter
+    def _indices_view(self, value):
+        if not hasattr(self, '_shown'):
+            # TODO: find a better solution for this. Needed to avoid race condition on layer instantiation
+            # if n_dimensional is True, both `shown` and `size` are needed to define _indices_view,
+            # and they rely on each other to initialize. See napari#3625 test failures.
+            self.__indices_view = value
+        else:
+            self.__indices_view = value[self.shown[value]]
```

Comment:
could you explain why `_shown` is not defined in the constructor?

Reply:
Even if I define it in the constructor as something like
```python
       self._shown = np.empty(0).astype(bool)
```
test fail because
```python
    @_indices_view.setter
    def _indices_view(self, value):
>       self.__indices_view = value[self.shown[value]]
E       IndexError: index 0 is out of bounds for axis 0 with size 0
```

I could not come up with anything better than `np.empty`; suggestions?

Reply:
The same will happen if you set data to the empty list. You should change this to at least checking the size of `selff._shown`, not checking if it is defined.

Reply:
yeah, make sense!